src/Microsoft.CodeAnalysis.BannedApiAnalyzers/UnitTests/SymbolIsBannedAnalyzerTests.cs

```diff
@@ -208,6 +210,46 @@ void M()
             await VerifyCSharpAnalyzerAsync(source, bannedText, GetCSharpResultAt(0, SymbolIsBannedAnalyzer.SymbolIsBannedRule, "Banned", ""));
         }
 
+        [Fact]
+        public async Task CSharp_BannedApi_MultipleFiles()
+        {
+            var source = @"
+namespace N
+{
+    class BannedA { }
+    class BannedB { }
+    class NotBanned { }
+    class C
+    {
+        void M()
+        {
+            var a = {|#0:new BannedA()|};
+            var b = {|#1:new BannedB()|};
+            var c = new NotBanned();
+        }
+    }
+}";
+
+            var test = new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources = { source },
+                    AdditionalFiles =
+                    {
+                        ("BannedSymbols.txt", @"T:N.BannedA") ,
+                        ("BannedSymbols.Other.txt", @"T:N.BannedB"),
```

Comment:
I think it would be worth to add a test where you have some symbols duplicated in the various files.

Reply:
I noticed something strange while adding the test: the `AdditionalFiles` collection is sorted by `DocumentId`, which is essentially a `Guid`, which means the order of the additional files is random.

This _may_ be a test environment artifact, but I don't think so. It looks like a real issue, since random file order doesn't make the build reproducible. Also, I found an issue which mentions this: https://github.com/dotnet/roslyn/issues/8605

In the case of this analyzer, it randomly swaps the diagnostic location with the additional location. Note that this PR doesn't really introduce this issue, since you can already reproduce it if you provide several additional files named `BannedSymbols.txt`, but from different directories.

How would you handle that?

Reply:
That's a really good remark. Pinging @mavasani here

In case of the banned symbols we should not end-up with contradictory configurations so a union/merge of all banned symbols would/should work.

Reply:
Yes, in any case the analyzer would take a merge of the rules, except that right now:
 - If the same symbol is present in multiple config files with a different description text, the RS0030 diagnostic (*Do not use banned APIs*) for this symbol will pick the description text randomly from one of the files.
 - The RS0031 diagnostic (*The list of banned symbols contains a duplicate*) will pick its location and additional location randomly.

I could add an `orderby additionalFile.Path` clause after the `where fileName != null ...` line in the analyzer to work around this and make it deterministic, I'm just not sure if I *should* since it looks like a Roslyn issue.

Reply:
Thanks, I didn't considered these.

I cannot make a call here so let's wait the input from a MS folk.

> * If the same symbol is present in multiple config files with a different description text, the RS0030 diagnostic (_Do not use banned APIs_) for this symbol will pick the description text randomly from one of the files.

We could "merge" the description, pick first (current) assuming the ordering issue will get fixed and maybe add this case to the analyzer I need to implement for invalid configurations https://github.com/dotnet/roslyn-analyzers/issues/3267.

Reply:
I would prefer that some diagnostic is given to the user indicating that they have duplicate entries.

Reply:
@jmarolf Ok, I added a test which validates that RS0031 is emitted in that case.

As stated previously, I had to add an `orderby` clause to make the behavior deterministic, and this was actually an existing issue since you already could provide multiple `BannedSymbols.txt` files from different directories.

Reply:
>  and this was actually an existing issue since you already could provide multiple BannedSymbols.txt files from different directories.

True, thanks for cleaning this up.

Reply:
I'm coming across this much later but... if the `AnalyzerOptions.AdditionalFiles` are coming in a nondeterministic order, that is a bug that we should file on the analyzer host/compiler.

Reply:
@RikkiGibson yes, indeed, it breaks determinism. I didn't open another issue since I considered it's already filled under #8605.